cmd/azure-config-credentials-injector/main.go

```diff
@@ -57,17 +63,26 @@ func mergeCloudConfig(_ *cobra.Command, args []string) error {
 		return fmt.Errorf("%s env variable should be set up", clientIDEnvKey)
 	}
 
-	azureClientSecret, found := os.LookupEnv(clientSecretEnvKey)
-	if !found {
-		return fmt.Errorf("%s env variable should be set up", clientSecretEnvKey)
+	// Handle error cases for authentication methods. We should configure exactly one method.
+	tenantId, federatedTokenFile, err := lookupWorkloadIdentityEnv(tenantIDEnvKey, federatedTokenEnvKey)
+	azureClientSecret, secretFound := os.LookupEnv(clientSecretEnvKey)
+	if err == nil && secretFound {
+		// Workload identity is available, but client secret is provided too.
+		// If we don't fail here we would use client secret unnecessarily, also we can assume that user intended to use workload identity because all required values are provided.
+		return fmt.Errorf("%s env variable is set along with env variables %s and %s which indicate workload identity auth should be used, this should never happen\n"+
+			"Please consider reporting a bug to CCO team: https://issues.redhat.com", clientSecretEnvKey, tenantIDEnvKey, federatedTokenEnvKey)
+	}
+	if err != nil && !secretFound {
+		// Workload identity failed, and we don't have a secret - can't continue.
+		return fmt.Errorf("user+secret method failed: %s is missing; workload identity method failed: %v", clientSecretEnvKey, err)
```

Comment:
would it be possible to have `err != nil && secretFound` ?

just wondering if we need to cover all cases

Reply:
With the conditions here we rule out the failure cases which are:

1) we have both secret and workload identity 
2) we have neither

So what remains is that we either have secret or we have workload identity. Later we resolve this in `prepareCloudConfig()` where we take one or the other, prioritizing workload identity. So all cases are covered.

So sure we can have the conditions explicitly here, but then I believe we'd have to refactor `prepareCloudConfig()` so it does not do any decisions (which it already does with `useManagedIdentityExtensionConfigKey`). That would make the code more readable perhaps by having the logic at one place - I was tempted to go that route, but we have to consider the time invested to refactoring and value added.